rasa/utils/tensorflow/models.py

```diff
@@ -388,180 +234,22 @@ def load(
         )
         # create empty model
         model = cls(*args, **kwargs)
+
         # need to train on 1 example to build weights of the correct size
-        model.fit(
-            model_data_example,
-            epochs=1,
-            batch_size=1,
-            evaluate_every_num_epochs=0,
-            evaluate_on_num_examples=0,
-            batch_strategy=SEQUENCE,
-            silent=True,  # don't confuse users with training output
-            loading=True,  # don't use tensorboard when doing a dummy fit run
-            eager=(
-                False if finetune_mode else True
-            ),  # load in eager mode only for prediction phase
-        )
+        model.compile(run_eagerly=False if finetune_mode else True)
+        data_generator = RasaBatchDataGenerator(model_data_example, batch_size=1)
+        model.fit(data_generator, verbose=False)
         # load trained weights
         model.load_weights(model_file_name)
 
+        if not finetune_mode:
+            # prepare the model for prediction
+            model.prepare_for_predict()
+            model.prepared_for_prediction = True
```

Comment:
I think the model also have to be run on prediction example to build the prediction graph, or did it somehow changed?

Reply:
It's not needed anymore. The prediction graph don't uses any new layers. All layers are already build on the `fit` call above. However, it might be a bit faster during inference if we predict already one example during loading. Will check that.

Reply:
Ok, predicting the first example using DIET takes 0.9635 sec vs 0.0632 sec. So I guess we should predict one example already during loading to speed things up.

Reply:
yeah, I think it's not about building actual layers, but rather converting python code into c++ executable tf graph